### PR TITLE
fix: mnist notebook for neo tensorflow

### PR DESCRIPTION
**Pull request by khkothar**
*Issue #, if available:*

*Description of changes:*
* Fix training script to save model in /opt/ml/model directory so that saved model is uploaded to s3 bucket
* Use new neo tensorflow image for deploying compiled models

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by ashishgupta023 — APPROVED** [review]

File: sagemaker_neo_compilation_jobs/tensorflow_distributed_mnist/tensorflow_distributed_mnist_neo.ipynb [hunk 1]
```diff
@@ -123,11 +119,12 @@
...
     "mnist_estimator = TensorFlow(entry_point='mnist.py',\n",
     "                             role=role,\n",
-    "                             framework_version='1.11.0',\n",
+    "                             framework_version='1.15.3',\n",
+    "                             py_version='py3',\n",
     "                             training_steps=1000, \n",
     "                             evaluation_steps=100,\n",
-    "                             train_instance_count=2,\n",
-    "                             train_instance_type='ml.c4.xlarge')\n",
+    "                             instance_count=2,\n",
```
Comment (ngluna) [inline sagemaker_neo_compilation_jobs/tensorflow_distributed_mnist/tensorflow_distributed_mnist_neo.ipynb:126]: `TypeError: __init__() got an unexpected keyword argument 'instance_count'`

File: sagemaker_neo_compilation_jobs/tensorflow_distributed_mnist/tensorflow_distributed_mnist_neo.ipynb [hunk 2]
```diff
@@ -123,11 +119,12 @@
...
+    "                             instance_type='ml.c4.xlarge')\n",
```
Comment (ngluna) [inline sagemaker_neo_compilation_jobs/tensorflow_distributed_mnist/tensorflow_distributed_mnist_neo.ipynb:127]: TypeError: __init__() got an unexpected keyword argument 'instance_type'

File: sagemaker_neo_compilation_jobs/tensorflow_distributed_mnist/tensorflow_distributed_mnist_neo.ipynb [hunk 3]
```diff
@@ -123,11 +119,12 @@
     "\n",
     "mnist_estimator = TensorFlow(entry_point='mnist.py',\n",
     "                             role=role,\n",
-    "                             framework_version='1.11.0',\n",
+    "                             framework_version='1.15.3',\n",
```
Comment (ngluna) [inline sagemaker_neo_compilation_jobs/tensorflow_distributed_mnist/tensorflow_distributed_mnist_neo.ipynb:122]: Why are we using TF v<2?

File: sagemaker_neo_compilation_jobs/tensorflow_distributed_mnist/tensorflow_distributed_mnist_neo.ipynb [hunk 4]
```diff
@@ -6,7 +6,7 @@
...
     "\n",
-    "This notebook can be compared to [TensorFlow MNIST distributed training notebook](https://github.com/awslabs/amazon-sagemaker-examples/blob/master/sagemaker-python-sdk/tensorflow_distributed_mnist/tensorflow_distributed_mnist.ipynb) in terms of its functionality. We will do the same classification task, but this time we will compile the trained model using the Neo API backend, to optimize for our choice of hardware. Finally, we setup a real-time hosted endpoint in SageMaker for our compiled model using the Neo Deep Learning Runtime."
+    "This notebook can be compared to [TensorFlow MNIST training and serving notebook](https://github.com/aws/amazon-sagemaker-examples/blob/master/sagemaker-python-sdk/tensorflow_script_mode_training_and_serving/tensorflow_script_mode_training_and_serving.ipynb) in terms of its functionality. We will do the same classification task, but this time we will compile the trained model using the Neo API backend, to optimize for our choice of hardware. Finally, we setup a real-time hosted endpoint in SageMaker for our compiled model using the Neo Deep Learning Runtime."
```
Comment (ngluna) [inline sagemaker_neo_compilation_jobs/tensorflow_distributed_mnist/tensorflow_distributed_mnist_neo.ipynb:9]: Replace with:
"In this notebook you will compile a trained model using Amazon SageMaker Neo. This notebook is similar to the TensorFlow MNIST training and serving notebook in that you will [INSERT]. You will complete the same classification task, however this time you will compile the trained model using the SageMaker Neo API on the backend. SageMaker Neo will optimize your model to run on your choice of hardware. At the end of this notebook you will setup a real-time hosting endpoint in SageMaker for your SageMaker Neo compiled model using the Neo Deep Learning Runtime (DLR)."

File: sagemaker_neo_compilation_jobs/tensorflow_distributed_mnist/tensorflow_distributed_mnist_neo.ipynb (hunk 1 above)
Comment (khkothar) [reply sagemaker_neo_compilation_jobs/tensorflow_distributed_mnist/tensorflow_distributed_mnist_neo.ipynb:126]: I'm using SageMaker SDK 2.x. This version is backwards incompatible with 1.x. Are these notebooks supposed to work with 1.x as well?

File: sagemaker_neo_compilation_jobs/tensorflow_distributed_mnist/tensorflow_distributed_mnist_neo.ipynb (hunk 2 above)
Comment (khkothar) [reply sagemaker_neo_compilation_jobs/tensorflow_distributed_mnist/tensorflow_distributed_mnist_neo.ipynb:127]: same as above

File: sagemaker_neo_compilation_jobs/tensorflow_distributed_mnist/tensorflow_distributed_mnist_neo.ipynb (hunk 3 above)
Comment (khkothar) [reply sagemaker_neo_compilation_jobs/tensorflow_distributed_mnist/tensorflow_distributed_mnist_neo.ipynb:122]: Neo currently doesn't support TF 2.x version. So, we're using 1.15 which is the current TF 1.x version.

**Review by ngluna — APPROVED** [review]

**Pull request merged**